### PR TITLE
Add ability to exclude directory and files

### PR DESCRIPTION
**Pull request by malukenho**
@weierophinney I was thinking about use `ignore` instead of `exclude`. Wdyt?

### CODE REVIEW

**Comment by malukenho** [comment]
@weierophinney to when you have time to review it 😊

File: src/Command/Checker.php [hunk 1]
```diff
@@ -47,18 +49,32 @@ protected function configure()
                 'directory',
                 InputArgument::IS_ARRAY | InputArgument::REQUIRED,
                 'Directory to scan *.php files'
+            )
+            ->addOption(
+                'exclude-dir',
+                null,
+                InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+                'Prevent directory to be used when check for doc headers'
```
Comment (weierophinney) [inline src/Command/Checker.php]: Maybe a better message would be: "Exclude the specified directory from being scanned; declare multiple directories with multiple invocations of this option."

File: src/Command/Checker.php [hunk 2]
```diff
@@ -47,18 +49,32 @@ protected function configure()
...
+            )
+            ->addOption(
+                'exclude',
+                null,
+                InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+                'Prevent file to be used when check for doc headers'
```
Comment (weierophinney) [inline src/Command/Checker.php]: Maybe a better message would be: "Exclude the specified file from being scanned; declare multiple files with multiple invocations of this option."

File: src/Command/Checker.php [hunk 3]
```diff
@@ -47,18 +49,32 @@ protected function configure()
...
+        $excludedDirectories = $input->getOption('exclude-dir') ?: [];
+        $excludedFiles       = $input->getOption('exclude') ?: [];
+        $finder              = (new IOResourcePathResolution($directory, $excludedDirectories, $excludedFiles))->__invoke();
+        $validator           = new RegExp($this->header);
 
         /* @var $file \Symfony\Component\Finder\SplFileInfo */
-        foreach ($finder as $directory) {
-            foreach ($directory as $file) {
+        foreach ($finder as $directoryList) {
+            foreach ($directoryList as $file) {
```
Comment (weierophinney) [inline src/Command/Checker.php]: Why the change in variable name to `$directoryList`? It's semantically less clear to me now what it represents, as iterating over it still produces files...

**Review by weierophinney — CHANGES_REQUESTED** [review]
Tests and implementation look good, and I verified them both against the console and finder usage docs.

I have made suggestions for better help messages for the new options, as well as asked a question about a variable name change; with those corrected, this looks great!

File: src/Command/Checker.php (hunk 3 above)
Comment (malukenho) [reply src/Command/Checker.php]: @weierophinney it had a previous variable called directory, I was wrongly overriding its content it here. what do you suggest?

Comment (weierophinney) [reply src/Command/Checker.php]: Ah, okay, that was not clear. Maybe `$dir` or `$fileList`, then?

File: src/Command/Checker.php [hunk 4]
```diff
@@ -54,13 +54,15 @@ protected function configure()
                 'exclude-dir',
                 null,
                 InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
-                'Prevent directory to be used when check for doc headers'
+                'Exclude the specified directory from being scanned; declare multiple directories ' .
+                'with multiple invocations of this option.'
```
Comment (weierophinney) [inline src/Command/Checker.php]: Usually I see concatentation operators on the _next_ line, to make it clear that the string is a continuation of the previous line; otherwise, it looks like you have an extra argument.

Otherwise, :+1: for the new message.

**Comment by weierophinney** [comment]
@malukenho Looks good -  I made a note about the placement of the concatenation operator in those broken-up description strings, though, as I think changing its position will make determining which lines begin arguments more clear.

**Review by weierophinney — APPROVED** [review]
:+1:
